panel/pane/vtk.py

```diff
@@ -0,0 +1,519 @@
+        _strips = _dump_data_array(scDirs, datasetDir, dataDir, dataset.GetStrips().GetData(), {}, compress)
+        _cells['strips'] = _strips
+        _cells['strips']['vtkClass'] = 'vtkCellArray'
+
+    _dump_color_array(scDirs, datasetDir, dataDir, colorArrayInfo, container, compress)
+
+    # # PointData TCoords
+    _dump_tcoords(scDirs, datasetDir, dataDir, dataset, container, compress)
+
+    return root
+
+
+_writer_mapping['vtkPolyData'] = _dump_poly_data
+
+
+def _dump_unstructured_grid(scDirs, datasetDir, dataDir, dataset, colorArrayInfo, root={}, compress=True):
+    geofilter = vtk.vtkGeometryFilter()
+    geofilter.SetInputData(dataset)
+    geofilter.Update()
+    polydata = geofilter.GetOutput()
+    del geofilter
+    return _dump_poly_data(scDirs, datasetDir, dataDir, polydata, colorArrayInfo, root={}, compress=True)
+
+
+_writer_mapping['vtkUnstructuredGrid'] = _dump_unstructured_grid
+
+
+def _dump_image_data(scDirs, datasetDir, dataDir, dataset, colorArrayInfo, root={}, compress=True):
+    root['vtkClass'] = 'vtkImageData'
+    container = root
+
+    container['spacing'] = dataset.GetSpacing()
+    container['origin'] = dataset.GetOrigin()
+    container['extent'] = dataset.GetExtent()
+
+    _dump_all_arrays(scDirs, datasetDir, dataDir, dataset, container, compress)
+
+    return root
+
+
+_writer_mapping['vtkImageData'] = _dump_image_data
+
+
+def _write_data_set(scDirs, dataset, colorArrayInfo, newDSName, compress=True):
+
+    dataDir = os.path.join(newDSName, 'data')
+
+    root = {}
+    root['metadata'] = {}
+    root['metadata']['name'] = newDSName
+
+    writer = _writer_mapping[dataset.GetClassName()]
+    if writer:
+        writer(scDirs, newDSName, dataDir, dataset, colorArrayInfo, root, compress)
+    else:
+        print(dataset.GetClassName(), 'is not supported')
+
+    scDirs.append([os.path.join(newDSName, 'index.json'), json.dumps(root, indent=2)])
+
+
+class Vtk(Bokeh):
+    """
+    Vtk panes allow rendering Vtk objects.
+    """
+
+    @classmethod
+    def _import_bk_model(cls):
+        if 'panel.models.vtk' not in sys.modules:
+            if isinstance(comm, JupyterComm):
+                cls.param.warning('VtkPlot was not imported on instantiation '
+                                   'and may not render in a notebook. Restart '
+                                   'the notebook kernel and ensure you load '
+                                   'it as part of the extension using:'
+                                   '\n\npn.extension(\'vtk\')\n')
+            from ..models.vtk import VtkPlot
+        else:
+            VtkPlot = getattr(sys.modules['panel.models.vtk'], 'VtkPlot')
+        return VtkPlot
+
+    def _get_model(self, doc, root=None, parent=None, comm=None):
+        """
+        Should return the bokeh model to be rendered.
+        """
+        self._import_bk_model()
+        return super()._get_model(doc, root, parent, comm)
```

Comment:
What actually calls ``model_from_render_window``? Would have expected that to be called here at least if the object isn't already a VtkPlot model.